packages/ember-extension-support/lib/data_adapter.js

```diff
@@ -163,7 +163,7 @@ export default EmberObject.extend({
 
   _nameToClass(type) {
     if (typeof type === 'string') {
-      type = getOwner(this)._lookupFactory(`model:${type}`);
+      type = getOwner(this).factoryFor(`model:${type}`).class;
```

Comment:
It seems there is some use of the types in Ember-Data. For example [here attributes are read off the class](https://github.com/emberjs/data/blob/c9d8212c857ca78218ad98d11621819b38dba98f/addon/-private/system/debug/debug-adapter.js#L32). Removing the double-extend means the `attributes` CP would be called and memoized on the base class, and thus shared between test runs etc.

This is definitely a change, described [in the design section](https://github.com/emberjs/rfcs/blob/master/text/0000-factoryFor.md#design-of-applicationinstancefactoryfor).

Some other examples of libraries this would impact are listed [in the drawbacks section](https://github.com/emberjs/rfcs/blob/master/text/0000-factoryFor.md#drawbacks).